flang/include/flang/Lower/Support/BoxValue.h

```diff
@@ -113,23 +113,36 @@ class AbstractArrayBox {
     return lbounds;
   }
 
+  /// If the entity is described by a box (e.g. it is a dummy, allocatable, or
+  /// pointer) The information in the source box that is also present in other
+  /// fields (e.g the extents) is the same, except the lower bounds of the
+  /// sourceBox that should be ignored.
+  mlir::Value getSourceBox() const { return sourceBox; }
```

Comment:
Is this the result of an embox or a box value in memory address?

Reply:
It can be both.
- If it is an an assumed shape dummy, it is the block argument fir.box
- If it is an allocatable tracked with a descriptor, it is the result of the fir.load fir.ref<fir.box>
- If it is an associated entity from an associate construct with an array section selector, it will (that's a TODO in Val PR) be the result of a fir.embox.